pkg/server/event_handler_test.go

```diff
@@ -613,11 +620,20 @@ func (m *testMessenger) Send(_ context.Context, data []byte, attr map[string]str
 	if m == nil {
 		return nil
 	}
-	err := protojson.Unmarshal(data, m.gotPmapEvent)
-	if err != nil {
-		return fmt.Errorf("failed to unmarshal to PmapEvent: %w", err)
+
+	// len(data) can be zero when generatePmapEventBytes() fails.
+	// if len(data) is zero, there is no point to Unmarshal data,
+	// the unmarshal here is to help test generatePmapEventBytes() generated
+	// the right eventBytes.
+	if len(data) != 0 {
```

Comment:
Why do you need this?

Reply:
Because when there is some error with yaml parsing, the data will be nil (have length of 0), and this will cause `protojson.Unmarshal()` to throw error.
See: https://go.dev/play/p/hHa3y_AC36c

Reply:
Why the data will be nil, I thought we will write some failure info into failure table.

Reply:
Discussed offline and explained why, updated the comment to better explain the usage of this function.

Reply:
Would pubsub fail if data is nil?

I think the comment here could be as simple as "Don't record gotPmapEvent when data is nil". The current comment requires too much context from other places. I can't even find where it defines `generatePmapEventBytes`

Reply:
Confirmed with @sailorlqh , pubsub won't fail. Here we need to do unmarshal in UT is because UT needs to verify the send pamp event.

Reply:
pubsub won't fail. See [here](https://cloud.google.com/pubsub/docs/reference/rest/v1/PubsubMessage#:~:text=A%20message%20that%20is%20published,differently%20depending%20on%20the%20language.)

If data is empty, as long as we have one attribute, publish won't fail. And we do have attribute here.